analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -437,6 +426,44 @@ public static ConditionalAccessExpressionSyntax GetRootConditionalAccessExpressi
             return current;
         }
 
+        // Copy of
+        // https://github.com/dotnet/roslyn/blob/575bc42589145ba18b4f1cc2267d02695f861d8f/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxNodeExtensions.cs#L347
+        public static bool IsLeftSideOfAssignExpression(this SyntaxNode node)
+            => node?.Parent is AssignmentExpressionSyntax { RawKind: (int)SyntaxKind.SimpleAssignmentExpression } assignment &&
+               assignment.Left == node;
```

Comment:
```suggestion
        public static bool IsLeftSideOfAssignExpression(this SyntaxNode node) =>
            node?.Parent is AssignmentExpressionSyntax { RawKind: (int)SyntaxKind.SimpleAssignmentExpression } assignment 
            && assignment.Left == node;
```

EDIT: I noticed that you kept exactly the code and style from Roslyn for all the methods, so maybe that's to be discussed [here](https://trello.com/c/nw3yLmEY/52-copying-internal-code-from-roslyn). If so ignore this comment